File: backend/app/schemas/editor.py
"""Pydantic-схемы для эндпоинта /editor."""
from __future__ import annotations

from pydantic import BaseModel, Field


class ElementInfo(BaseModel):
    editable_id: str
    file_path: str
    element_html: str


class EditElementRequest(BaseModel):
    project_id: str
    element: ElementInfo
    instruction: str = Field(..., min_length=1, max_length=2000)
    ai_model: str = Field(default="gpt-5.4")


class EditElementResponse(BaseModel):
    task_id: str
    project_id: str
    file_path: str
    status: str = "queued"


class GetFileRequest(BaseModel):
    project_id: str
    file_path: str


class UpdateFileRequest(BaseModel):
    project_id: str
    file_path: str
    content: str

File: frontend/src/api/editor.ts
import apiClient from './client';

export interface FileContent {
  path: string;
  content: string;
}

export interface AIEditRequest {
  project_id: string;
  element: {
    editable_id: string;
    file_path: string;
    element_html: string;
  };
  instruction: string;
  ai_model?: string;
}

export interface AIEditResponse {
  task_id: string;
}

export async function listFiles(projectId: string): Promise<string[]> {
  const { data } = await apiClient.get<{ project_id: string; files: string[] }>('/editor/files', {
    params: { project_id: projectId },
  });
  return data.files;
}

export async function getFile(projectId: string, path: string): Promise<FileContent> {
  const { data } = await apiClient.get<{ project_id: string; file_path: string; content: string }>('/editor/file', {
    params: { project_id: projectId, file_path: path },
  });
  return { path: data.file_path, content: data.content };
}

export async function saveFile(projectId: string, path: string, content: string): Promise<void> {
  await apiClient.put('/editor/file', { project_id: projectId, file_path: path, content });
}

export async function submitAIEdit(payload: AIEditRequest): Promise<AIEditResponse> {
  const { data } = await apiClient.post<AIEditResponse>('/editor/edit', payload);
  return data;
}

export async function triggerRebuild(projectId: string): Promise<void> {
  await apiClient.post('/editor/rebuild', null, { params: { project_id: projectId } });
}
